x/ecocredit/server/core/bridge_receive.go

```diff
@@ -89,7 +89,8 @@ func (k Keeper) BridgeReceive(ctx context.Context, req *core.MsgBridgeReceive) (
 		if err != nil {
 			return nil, err
 		}
-		if batch.Metadata == req.BatchMetadata {
+		// the timestamp stored in the batch is more granular than the date in the denom representation, so we match here.
+		if batch.StartDate.AsTime().Equal(*req.StartDate) && batch.EndDate.AsTime().Equal(*req.EndDate) && batch.Metadata == req.BatchMetadata {
```

Comment:
@technicallyty The current metadata converter rounds to the nearest day, so if the YYYYMMDD version matches then this will match. 

Why do we need to compare the metadata? If we're already selecting the batch based on project ID and date range, then I don't believe we need to match on metadata?

Reply:
hmm I did this with the idea that there might be multiple batches with same start/end dates within a project but perhaps that is not possible 🧐

Reply:
I think it's unlikely but possible and the extra check would be good to have. :+1:

Reply:
Oof. Based on conversations in slack, it looks like there might not be any batch metadata to fill...

Reply:
I think we should still check and leave this as is for now. The larger issue is how we populate the batch metadata that is currently blank in vintages stored in the toucan contracts.